rasa_sdk/types.py

```diff
@@ -0,0 +1,33 @@
+from typing import Any, Dict, List, Optional, Text, Union
+
+from typing_extensions import TypedDict
+
+
+class TrackerState(TypedDict):
```

Comment:
should we rather use pythons new data classes? (and a backport for 3.6 https://pypi.org/project/dataclasses/)

Reply:
I think that could be better indeed, but I do not know enough about the specifics of the tracker state / action call:
- would renaming the method `from_dict()` to `from_dataclass()` be acceptable?
- is the type definition I made here valid? (I marked some elements required, before they were optional)

Reply:
after thinking a bit more about it, I'm not sure that we would benefit from the dataclass here (vs keeping the TypedDict). And if we wanted to do this it might change the public API of the SDK. For instance, instead of calling custom action with plain dicts, we would call them with dataclasses. And I'm not sure it's worth the trouble. WDYT?

Reply:
ok that sounds reasonable, let's stick with the typeddicts :+1: